src/crypto/aead/aead_macros.rs

```diff
@@ -4,7 +4,8 @@ macro_rules! aead_module (($seal_name:ident,
                            $open_detached_name:ident,
                            $keybytes:expr,
                            $noncebytes:expr,
-                           $tagbytes:expr) => (
+                           $tagbytes:expr,
+                           $test_gate:expr) => (
```

Comment:
I'd rather have `target_feature = "aes"` for aes tests

Reply:
I'm not sure how that would work unless you wanted to expand the test matrix to do a separate AES and non AES build, in which case we're back to the earlier discussion about packaging and reproducibility.

Reply:
Instead of fixing tests for aes only (`crate::init().is_ok() && is_available()`) you added $test_gate variable for all modules. Am I missing something?

Reply:
The goal was to keep the unit tests consistent across all of the internal implementations of aead. The only way to do that is introduce a gate that checks preconditions (runtime support in this case). There are a few ways to factor it, but this seemed like the most minor surgery.

Reply:
Personally, I really like this way of keeping the amount of boiler plate as small as possible without introducing any complications into the build.

The only other reasonable option I see would be to require the unit tests to be run on a machine with support for hardware AES which does sound entirely reasonable considering that they wont run with 16MB of RAM or take unpratically long on a 33 MHz CPU as well.

Reply:
@kpp are we good on this point?

Reply:
yes.